app/models/mixins/relationship_mixin.rb

```diff
@@ -653,6 +653,14 @@ def replace_children(*child_objs)
     child_objs = child_objs.flatten
     return remove_all_children if child_objs.empty?
 
+    if use_ancestry?
+      to_add = child_objs.collect(&:id) - children.collect(&:id)
+      to_add, to_keep = child_objs.partition { |c| to_add.include?(c.id) }
+      (children - to_keep).each { |child| remove_children(child) }
```

Comment:
`#remove_children` and `#add_children` already work on sets of children, could this just be `remove_children(to_remove)` and `add_children(to_add_)`?